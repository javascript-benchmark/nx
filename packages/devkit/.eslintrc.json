{
  "extends": "../../.eslintrc",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "@typescript-eslint/explicit-module-boundary-types": ["error"],
        "@typescript-eslint/no-restricted-imports": [
          "error",
          {
            "paths": [
              "@nx/workspace",
              "@angular-devkit/core",
              "@angular-devkit/architect",
              "@angular-devkit/schematics"
            ],
            "patterns": [
              {
                "group": ["nx/**/*"],
                "message": "Use requireNx() from packages/devkit/nx.ts OR use a type import instead.",
                "allowTypeImports": true
              },
              {
                "group": ["@nx/devkit/**/*"],
                "message": "Use a relative import"
              }
            ]
          }
        ]
      }
    },
    {
      "files": ["*.spec.ts"],
      "rules": {
        "@typescript-eslint/no-restricted-imports": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["./package.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/nx-plugin-checks": "error"
      }
    }
  ]
}
