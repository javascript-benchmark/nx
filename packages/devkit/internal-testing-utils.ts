/* eslint-disable @typescript-eslint/no-restricted-imports */
export * from 'nx/internal-testing-utils/assert-valid-migrations';
export * from 'nx/internal-testing-utils/run-migration-against-this-workspace';
export * from 'nx/internal-testing-utils/with-environment';
