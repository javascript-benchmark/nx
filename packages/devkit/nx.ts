// After Nx v18, this can be removed and replaced with either:
// - import {} from 'nx/src/devkit-exports'
// - import {} from 'nx/src/devkit-internals'
export function requireNx(): typeof import('nx/src/devkit-exports') &
  Partial<typeof import('nx/src/devkit-internals')> {
  try {
    let result = { ...require('nx/src/devkit-exports') };
    try {
      result = {
        ...result,
        // Remove in Nx v18, devkit should not support Nx v16.0.2 at that point.
        ...require('nx/src/devkit-internals'),
      };
    } catch {}
    return result;
  } catch {
    // Remove in Nx V17, devkit should not support Nx < 16 at that point.
    return require('./nx-reexports-pre16');
  }
}
