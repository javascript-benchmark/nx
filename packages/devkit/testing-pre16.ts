/* eslint-disable @typescript-eslint/no-restricted-imports */
export {
  createTreeWithEmptyWorkspace,
  createTreeWithEmptyV1Workspace,
} from 'nx/src/generators/testing-utils/create-tree-with-empty-workspace';
export { createTree } from 'nx/src/generators/testing-utils/create-tree';
