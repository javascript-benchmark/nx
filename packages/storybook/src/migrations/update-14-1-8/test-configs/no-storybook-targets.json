{
  "projects": {
    "ui-one": {
      "projectType": "library",
      "root": "libs/ui/one",
      "sourceRoot": "libs/ui/one/src",
      "targets": {}
    },
    "ui-two": {
      "projectType": "library",
      "root": "libs/ui/two",
      "sourceRoot": "libs/ui/two/src",
      "targets": {}
    },
    "ui-three": {
      "projectType": "library",
      "root": "libs/ui/three",
      "sourceRoot": "libs/ui/three/src",
      "targets": {}
    },
    "main-app": {
      "projectType": "application",
      "root": "apps/main-app",
      "sourceRoot": "apps/main-app/src",
      "prefix": "katst",
      "targets": {
        "build": {
          "executor": "@angular-devkit/build-angular:browser",
          "outputs": ["{options.outputPath}"]
        }
      }
    }
  }
}
