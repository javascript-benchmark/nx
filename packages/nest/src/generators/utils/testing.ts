import { addProjectConfiguration, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

export function createTreeWithNestApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  addProjectConfiguration(tree, appName, {
    root: `apps/${appName}`,
    sourceRoot: `apps/${appName}/src`,
    projectType: 'application',
    targets: {},
  });
  return tree;
}
