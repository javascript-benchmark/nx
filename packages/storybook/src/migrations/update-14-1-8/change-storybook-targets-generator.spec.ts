import { addProjectConfiguration, readNxJson, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import changeStorybookTargetsGenerator from './change-storybook-targets-generator';
import * as defaultConfig from './test-configs/default-config.json';
import * as customNames from './test-configs/custom-names-config.json';
import * as nonAngular from './test-configs/non-angular.json';
import * as extraOptions from './test-configs/extra-options-for-storybook.json';
import * as noStorybookBuildTarget from './test-configs/no-build-storybook-target.json';
import * as noStorybook from './test-configs/no-storybook-targets.json';

describe('Change the Storybook targets for Angular projects to use native Storybooke executor', () => {
  let tree: Tree;

  describe('for all types of angular projects - non-buildable and buildable libs/apps', () => {
    beforeEach(async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    });

    it(`should set the browserTarget correctly in the Storybook config according to the type of project`, async () => {
      writeConfig(tree, defaultConfig);
      await changeStorybookTargetsGenerator(tree);
      expect(readNxJson(tree)).toMatchSnapshot();
    });

    it(`should set the browserTarget correctly even if target names are not the default`, async () => {
      writeConfig(tree, customNames);
      await changeStorybookTargetsGenerator(tree);
      expect(readNxJson(tree)).toMatchSnapshot();
    });

    it(`should keep any extra options added in the target`, async () => {
      writeConfig(tree, extraOptions);
      await changeStorybookTargetsGenerator(tree);
      expect(readNxJson(tree)).toMatchSnapshot();
    });
    it(`should work even if build-storybook does not exist`, async () => {
      writeConfig(tree, noStorybookBuildTarget);
      await changeStorybookTargetsGenerator(tree);
      expect(readNxJson(tree)).toMatchSnapshot();
    });

    it(`should not throw an error if no Storybook exists`, async () => {
      writeConfig(tree, noStorybook);
      await changeStorybookTargetsGenerator(tree);
      expect(readNxJson(tree)).toMatchSnapshot();
    });
  });

  describe('for non-angular projects', () => {
    it(`should not change their Storybook targets`, async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
      writeConfig(tree, nonAngular);
      await changeStorybookTargetsGenerator(tree);
      expect(readNxJson(tree)).toMatchSnapshot();
    });
  });
});

function writeConfig(tree: Tree, config: any) {
  Object.keys(config.projects).forEach((project) => {
    addProjectConfiguration(tree, project, config.projects[project]);
  });
}
