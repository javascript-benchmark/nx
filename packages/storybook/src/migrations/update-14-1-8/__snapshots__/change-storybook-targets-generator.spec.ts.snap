// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change the Storybook targets for Angular projects to use native Storybooke executor for all types of angular projects - non-buildable and buildable libs/apps should keep any extra options added in the target 1`] = `
{
  "affected": {
    "defaultBase": "main",
  },
  "npmScope": "proj",
  "tasksRunnerOptions": {
    "default": {
      "options": {
        "cacheableOperations": [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "nx/tasks-runners/default",
    },
  },
}
`;

exports[`Change the Storybook targets for Angular projects to use native Storybooke executor for all types of angular projects - non-buildable and buildable libs/apps should not throw an error if no Storybook exists 1`] = `
{
  "affected": {
    "defaultBase": "main",
  },
  "npmScope": "proj",
  "tasksRunnerOptions": {
    "default": {
      "options": {
        "cacheableOperations": [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "nx/tasks-runners/default",
    },
  },
}
`;

exports[`Change the Storybook targets for Angular projects to use native Storybooke executor for all types of angular projects - non-buildable and buildable libs/apps should set the browserTarget correctly even if target names are not the default 1`] = `
{
  "affected": {
    "defaultBase": "main",
  },
  "npmScope": "proj",
  "tasksRunnerOptions": {
    "default": {
      "options": {
        "cacheableOperations": [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "nx/tasks-runners/default",
    },
  },
}
`;

exports[`Change the Storybook targets for Angular projects to use native Storybooke executor for all types of angular projects - non-buildable and buildable libs/apps should set the browserTarget correctly in the Storybook config according to the type of project 1`] = `
{
  "affected": {
    "defaultBase": "main",
  },
  "npmScope": "proj",
  "tasksRunnerOptions": {
    "default": {
      "options": {
        "cacheableOperations": [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "nx/tasks-runners/default",
    },
  },
}
`;

exports[`Change the Storybook targets for Angular projects to use native Storybooke executor for all types of angular projects - non-buildable and buildable libs/apps should work even if build-storybook does not exist 1`] = `
{
  "affected": {
    "defaultBase": "main",
  },
  "npmScope": "proj",
  "tasksRunnerOptions": {
    "default": {
      "options": {
        "cacheableOperations": [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "nx/tasks-runners/default",
    },
  },
}
`;

exports[`Change the Storybook targets for Angular projects to use native Storybooke executor for non-angular projects should not change their Storybook targets 1`] = `
{
  "affected": {
    "defaultBase": "main",
  },
  "npmScope": "proj",
  "tasksRunnerOptions": {
    "default": {
      "options": {
        "cacheableOperations": [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "nx/tasks-runners/default",
    },
  },
}
`;
